### PR TITLE
Fix modules

### PR DESCRIPTION
**Pull request by iamstarkov**
#18

### CODE REVIEW

**Comment by iamstarkov** [comment]
@kinday @yakhinvadim ready for review

File: src/graph.js [hunk 1]
```diff
@@ -1,43 +1,51 @@
...
-import contract from './utils/contract';
+import { contractP } from './utils/contract';
 import { log } from './utils/console-methods';
 import { all, resolve } from './utils/promise-methods';
-import { normalize, relative as _relative, join as _join, dirname } from 'path';
+import { relative as _relative, join as _join, dirname } from 'path';
+import { isModule, isLocalFile } from './is-module';
 
-const debug = msg => R.tap(R.partial(log, [`\n${msg}\n`]));
+const d = msg => R.tap(R.partial(log, [msg]));
```
Comment (kinday) [inline src/graph.js]: This one should be hidden in utils.

File: src/graph.js [hunk 2]
```diff
@@ -1,43 +1,51 @@
...
-  const basedir = dirname(file);
-  const visitedAndCurrent = R.append(file, visited);
   return R.unless(
     R.contains(R.__, visited),
     R.pipeP(resolve,
       esDeps,
       R.map(R.pipe(
-        join(basedir),
-        resolveFile
+        R.when(isModule, R.identity),
```
Comment (kinday) [inline src/graph.js]: What’s the purpose of that line? Seems nonsensical to me: in both cases it works as `R.identity`.

File: src/graph.js (hunk 1 above)
Comment (iamstarkov) [inline src/graph.js]: thats true

File: src/graph.js (hunk 2 above)
Comment (iamstarkov) [inline src/graph.js]: you are right, in development i used this line as clear sign that `isModule` items are changed, and on other line `isLocalFile` are changed. will remove this line

File: src/utils/contract.js [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
-    new TypeError(`\`${name}\` should be \`${R.type(ctor())}\`, but got \`${R.type(param)}\``)
-  )
+  () => { throw new TypeError(errorText(name, ctor, param)); }
 )(param));
 
-export default contract;
+const contractP = R.curry((name, ctor, param) => R.unless(
+  R.is(ctor),
+  () => {
+    // console.log('meow');
```
Comment (iamstarkov) [inline src/utils/contract.js]: lol

File: src/graph.js (hunk 1 above)
Comment (iamstarkov) [inline src/graph.js]: extracted into new file

File: src/utils/contract.js (hunk 3 above)
Comment (iamstarkov) [inline src/utils/contract.js]: removed

File: src/graph.js (hunk 2 above)
Comment (iamstarkov) [inline src/graph.js]: fixed

**Pull request merged**